sql/core/src/test/scala/org/apache/spark/sql/connector/AlterTableTests.scala

```diff
@@ -328,7 +336,7 @@ trait AlterTableTests extends SharedSparkSession with QueryErrorsBase {
   }
 
   test("SPARK-39383 DEFAULT columns on V2 data sources with ALTER TABLE ADD/ALTER COLUMN") {
-    withSQLConf(SQLConf.DEFAULT_COLUMN_ALLOWED_PROVIDERS.key -> s"$v2Format, ") {
+    withSQLConf(SQLConf.DEFAULT_COLUMN_ALLOWED_PROVIDERS.key -> s"$v2Format,$catalog") {
```

Comment:
How does this conf affect the testing v2 in-memory catalog? I thought it's only for v1 file source.

Reply:
I think that it is because https://github.com/apache/spark/pull/50044/files#r1968081855

Reply:
@viirya is correct. We previously passed `"" ` as provider in the in-memory connector, which required workarounds like this. No longer needed as we pass the catalog name as provider. Simplifies testing.